tools/egghunter.rb

```diff
@@ -0,0 +1,150 @@
+require 'optparse'
+
+module Egghunter
+  class OptsConsole
+    def self.parse(args)
+      options = {}
+      parser = OptionParser.new do |opt|
+        opt.banner = "Usage: #{__FILE__} [options]\nExample: #{__FILE__} -f python -e W00T"
+        opt.separator ''
+        opt.separator 'Specific options:'
+
+        options[:badchars] = ''
+        options[:platform] = 'windows'
+        options[:arch]     = ARCH_X86 # 'x86'
+
+        opt.on('-f', '--format <String>', "See --list-formats for a list of supported output formats") do |v|
+          options[:format] = v
+        end
+
+        opt.on('-b', '--badchars <String>', "(Optional) Bad characters to avoid for the egg") do |v|
+          options[:badchars] = v
+        end
+
+        opt.on('-e', '--egg <String>', "The egg (Please give 4 bytes)") do |v|
+          options[:eggtag] = v
+        end
+
+        opt.on('-p', '--platform <String>', "(Optional) Platform") do |v|
+          options[:platform] = v
+        end
+
+        opt.on('--startreg <String>', "(Optional) The starting register") do |v|
+          # Do not change this key. This should matching the one in Rex::Exploitation::Egghunter
+          options[:startreg] = v
+        end
+
+        opt.on('--forward', "(Optional) To search forward") do |v|
+          # Do not change this key. This should matching the one in Rex::Exploitation::Egghunter
+          options[:startreg] = true
+        end
+
+        opt.on('--depreg <String>', "(Optional) The DEP register") do |v|
+          # Do not change this key. This should matching the one in Rex::Exploitation::Egghunter
+          options[:depreg] = v
+        end
+
+        opt.on('--depdest <String>', "(Optional) The DEP destination") do |v|
+          # Do not change this key. This should matching the one in Rex::Exploitation::Egghunter
+          options[:depdest] = v
+        end
+
+        opt.on('--depsize <Fixnum>', "(Optional) The DEP size") do |v|
+          # Do not change this key. This should matching the one in Rex::Exploitation::Egghunter
+          options[:depsize] = v
+        end
+
+        opt.on('--depmethod <String>', "(Optional) The DEP method to use (virtualprotect/virtualalloc/copy/copy_size)") do |v|
+          # Do not change this key. This should matching the one in Rex::Exploitation::Egghunter
+          options[:depmethod] = v
+        end
+
+        opt.on('-a', '--arch <String>', "(Optional) Architecture") do |v|
+          # Although this is an option, this is currently useless because we don't have x64 egghunters
+          options[:arch] = v
+        end
+
+        opt.on('--list-formats', "List all supported output formats") do
+          options[:list_formats] = true
+        end
+
+        opt.on_tail('-h', '--help', 'Show this message') do
+          $stdout.puts opt
+          exit
+        end
+      end
+
+      parser.parse!(args)
+
+      if options.empty?
+        raise OptionParser::MissingArgument, 'No options set, try -h for usage'
```

Comment:
when running without options the wrong message is displayed
`[x] missing argument: -f is required`